src/components/text-editor/__internal__/utils/utils.ts

```diff
@@ -168,20 +188,25 @@ export const getSelectedLength = (value) => {
   return length;
 };
 
-export function hasBlockStyle(value, type) {
+export function hasBlockStyle(value: EditorState, type: BlockType) {
   const { blockType } = getContentInfo(value);
   return blockType === type;
 }
 
-export function hasInlineStyle(value, style) {
+export function hasInlineStyle(value: EditorState, style: InlineStyleType) {
   return value.getCurrentInlineStyle().has(style);
 }
 
-export function isASCIIChar(str) {
+export function isASCIIChar(str: string) {
   return /^\S+$/.test(str);
 }
 
-export function replaceText(editorState, text, inlineStyle, forceSelection) {
+export function replaceText(
+  editorState: EditorState,
+  text: string,
+  inlineStyle: DraftInlineStyle
+  // forceSelection
```

Comment:
commented out, I think we need to force selection here as this is a workaround to an issue that occurs when you hit double space and a `.` is injected by mac OS, we can probably just push it on to the state rather than passing it in as arg etc